src/main/java/uk/gov/hmcts/reform/coh/schedule/trigger/QuestionRoundDeadlineElapsed.java

```diff
@@ -42,29 +40,22 @@ public void execute() {
         log.info(String.format("Executing %s", this.getClass()));
 
         Calendar calendar = new GregorianCalendar();
-        Optional<QuestionState> issuedState = stateService.retrieveQuestionStateByStateName(ISSUED.getStateName());
-        if (!issuedState.isPresent()) {
-            log.error(String.format("Unable to find question state: %s", ISSUED.getStateName()));
-            return;
-        }
-
-        Optional<QuestionState> elapsedState = stateService.retrieveQuestionStateByStateName(DEADLINE_ELAPSED.getStateName());
-        if (!elapsedState.isPresent()) {
-            log.error(String.format("Unable to find question state: %s", DEADLINE_ELAPSED.getStateName()));
-            return;
-        }
+
+        QuestionState issuedState = getQuestionStateByStateName(ISSUED);
+        QuestionState grantedState = getQuestionStateByStateName(QUESTION_DEADLINE_EXTENSION_GRANTED);
```

Comment:
Maybe it would be better to have something like this which uses Spring repo and minimised the number of DB calls. `List<QuestionState> validStates = findAllByQuestionStateIn(Arrays.asList(ISSUED, QUESTION_DEADLINE_EXTENSION_GRANTED);
QuestionState elapsedState = getQuestionStateByStateName(DEADLINE_ELAPSED);
`